griptape/drivers/tool/griptape_cloud_tool_driver.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+
+import os
+from types import MethodType
+from typing import TYPE_CHECKING, Any, Callable
+from urllib.parse import urljoin
+
+import requests
+from attrs import Factory, define, field
+from schema import Literal, Optional, Schema
+
+from griptape.artifacts import BaseArtifact, TextArtifact
+from griptape.drivers.tool.base_tool_driver import BaseToolDriver
+from griptape.utils.decorators import activity
+
+if TYPE_CHECKING:
+    from griptape.tools.base_tool import BaseTool
+
+
+@define
+class GriptapeCloudToolDriver(BaseToolDriver):
+    """Driver for interacting with tools hosted on the Griptape Cloud.
+
+    Attributes:
+        base_url: Base URL of the Griptape Cloud.
+        api_key: API key for the Griptape Cloud.
+        tool_id: ID of the tool to interact with.
+        headers: Headers to use for requests.
+        raise_on_same_activity: Whether to raise an error if an activity with the same name already exists on the tool.
+    """
+
+    base_url: str = field(default=Factory(lambda: os.getenv("GT_CLOUD_BASE_URL", "https://cloud.griptape.ai")))
+    api_key: str = field(default=Factory(lambda: os.environ["GT_CLOUD_API_KEY"]))
+    tool_id: str = field(kw_only=True)
+    headers: dict = field(
+        default=Factory(lambda self: {"Authorization": f"Bearer {self.api_key}"}, takes_self=True),
+        init=False,
+    )
+    raise_on_same_activity: bool = field(default=True, kw_only=True)
```

Comment:
maybe this should default to `False`? or named something like `overwrite_activties`? basically i think the default behavior should be overwriting the existing locally-defined activities but there should be a way to toggle that off. sorry not super clear

Reply:
Oh yeah, duh. I still think `raise_on_same_activity` is a good name though.

Reply:
it is no more